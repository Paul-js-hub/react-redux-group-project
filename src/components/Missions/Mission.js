import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import Badge from 'react-bootstrap/Badge';
import { joinMission, leaveMission } from '../../redux/missions/missions';
import styles from './missions.module.css';

const Mission = ({
  name, description, id, reserved,
}) => {
  const dispatch = useDispatch();
  const updateMission = () => {
    dispatch(joinMission(id));
  };

  const cancelMission = () => {
    dispatch(leaveMission(id));
  };
  return (
    <tbody key={id}>
      <tr>
        <td>{name}</td>
        <td>{description}</td>
        <td className={styles.reserved}>{reserved ? <Badge bg="primary">Active Member</Badge> : <Badge bg="secondary">NOT A MEMBER</Badge>}</td>
        <td className={styles.reserved}>{reserved ? <Badge className={styles.btnbadge} bg="danger" type="button" onClick={cancelMission}>Leave Mission</Badge> : <Badge className={styles.btnbadge} bg="secondary" type="button" onClick={updateMission}>Join Mission</Badge>}</td>
      </tr>
    </tbody>
  );
};

Mission.defaultProps = {
  id: PropTypes.string,
  name: PropTypes.string,
  description: PropTypes.string,
  reserved: PropTypes.bool,
};

Mission.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string,
  description: PropTypes.string,
  reserved: PropTypes.bool,
};

export default Mission;
