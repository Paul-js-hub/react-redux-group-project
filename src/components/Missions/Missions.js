const Missions = () => (
  <>
    Missions
  </>
);

export default Missions;
