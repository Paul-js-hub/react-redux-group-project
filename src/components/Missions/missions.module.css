.table {
  margin: 8px auto !important;
}

.reserved {
  vertical-align: middle !important;
}

.btnbadge {
  width: 95px;
  height: 28px;
}
