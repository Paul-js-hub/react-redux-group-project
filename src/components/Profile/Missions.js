import { useSelector } from 'react-redux';
import Card from 'react-bootstrap/Card';

const Missions = () => {
  const missions = useSelector((state) => state.missions.missions);
  const joinedMissions = missions.filter((mission) => mission.reserved === true);
  return (
    <div>

      {joinedMissions.map((mission) => (
        <div key={mission.mission_id}>
          <Card body>{mission.name}</Card>
        </div>
      ))}
    </div>
  );
};

export default Missions;
