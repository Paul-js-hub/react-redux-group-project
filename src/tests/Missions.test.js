import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Missions from '../components/Missions/Missions';
import store from '../redux/configureStore';

test('it returns a missions component', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Missions />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
