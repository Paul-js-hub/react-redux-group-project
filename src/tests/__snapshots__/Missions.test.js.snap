// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it returns a missions component 1`] = `
<div
  className="container"
>
  <table
    className="table table table-sm table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
      </tr>
    </thead>
  </table>
</div>
`;
